idem_azurerm/states/azurerm/web/function_app.py

```diff
@@ -0,0 +1,571 @@
+        {"name": "SCM_DO_BUILD_DURING_DEPLOYMENT", "value": "false"},
+    ]
+
+    if not isinstance(connection_auth, dict):
+        if ctx["acct"]:
+            connection_auth = ctx["acct"]
+        else:
+            ret[
+                "comment"
+            ] = "Connection information must be specified via acct or connection_auth dictionary!"
+            return ret
+
+    # Ensures location is specified
+    if "location" not in kwargs:
+        rg_props = await hub.exec.azurerm.resource.group.get(
+            ctx, resource_group, **kwargs
+        )
+
+        if "error" in rg_props:
+            log.error("Unable to determine location from resource group specified.")
+            return {
+                "error": "Unable to determine location from resource group specified."
+            }
+        kwargs["location"] = rg_props["location"]
+
+    # Handle storage account validation
+    if not storage_rg:
+        storage_rg = resource_group
+
+    storage_acct = await hub.exec.azurerm.storage.account.get_properties(
+        ctx, name=storage_account, resource_group=storage_rg
+    )
+
+    if "error" in storage_acct:
+        log.error(
+            "The storage account does not exist within the specified resource group."
+        )
+        ret[
+            "comment"
+        ] = "The storage account does not exist within the specified resource group."
+        return ret
+
+    # Ensure that the file path contains a zip file
+    filename = os.path.basename(functions_file_path)
+    if not ".zip" in filename.lower():
+        log.error(
+            "The specified file in functions_file_path is not a compressed (zip) file."
+        )
+        ret[
+            "comment"
+        ] = "The specified file in functions_file_path is not a compressed (zip) file."
+        return ret
+
+    # Set reserved for the ASP and Function App based upon OS type
+    if os_type.lower() == "windows":
+        reserved = False
+    else:  # linux
+        reserved = True
+
+    # Handle App Service Plan creation
+    if not app_service_plan:
+        app_service_plan = f"ASP-{name}"
+
+    plan = await hub.exec.azurerm.web.app_service_plan.get(
+        ctx, name=app_service_plan, resource_group=resource_group
+    )
+
+    if "error" in plan:
+        plan = await hub.exec.azurerm.web.app_service_plan.create_or_update(
+            ctx,
+            name=app_service_plan,
+            resource_group=resource_group,
+            kind="functionapp",
+            reserved=reserved,
+            sku="Y1",
+            **connection_auth,
+        )
+
+        if "error" in plan:
+            log.error(
+                f"Unable to create the App Service Plan {app_service_plan} in the resource group {resource_group}."
+            )
+            ret[
+                "comment"
+            ] = f"Unable to create the App Service Plan {app_service_plan} in the resource group {resource_group}."
+            return ret
+    else:
```

Comment:
if you have an `else` and then a single `if` underneath with no other logic, maybe make this an `elif`